### DISCUSSION TITLE
Dynamic Links - Different behaviors depending on the applications that open the link

### DISCUSSION THREAD

**Discussion by vikjovanov** [invertase/react-native-firebase#4585] [Q&A] [answered]
Hi,

Depending on the application that opens my link, the OS and whether or not I have enabled "forceRedirection", there are different behaviors that I try to understand.

Here is my table of how it works on iOS when it is an iOS that opens the links on the different apps.

![Screenshot 2020-11-21 at 15 03 06](https://user-images.githubusercontent.com/43797348/99879185-64108a00-2c0b-11eb-9d87-00a231ef8091.png)

As you can see, depending on whether I enable the efr or not, as soon as an application opens my link in an internal webview, the link seems to be lost (Snap, Insta, Facebook, Messenger, Twitter). On the other hand, if I disable the efr, the internal webview opens the Firebase preview with the "Open" button, and from there, when I click on Open, the link is not lost.

**NOK** = The app opens if installed / The store opens if app not installed but impossible to retrieve the link in the application.
**OK** = App opens if installed / Store opens if app not installed and the link is correctly retrieved in the application
**NOK (only redirect to the store)** = Whether the app is installed or not, the link only redirects to the store and even if I open/install the app from the store, I don't retrieve the link in the app.

Then here's another one, but this time the link is opened from an android.

![Screenshot 2020-11-21 at 15 19 36](https://user-images.githubusercontent.com/43797348/99879382-0715d380-2c0d-11eb-94dd-718129c393a3.png)

The first thing I notice on android is that even when it opens from a webview, the link doesn't seem to be "lost".
Then, another observation, whether I activate the efr or not, whatever application opens my link, there's never the preview page proposed by Firebase (maybe it's normal behavior, I can't find the info or I'm not looking well either).
On the other hand, on Snapchat Android, no matter how the link is shared, it always opens the store and the link is always lost.

So I have several questions:
- Why is it that on iOS, the link seems to get lost when it's opened in an internal webview? But when I use the firebase preview (with efr = 0), the link is not lost all of a sudden ?
Also, why is it that when the app is in background, the link is not lost ?
- How is it that on Android, the dynamics link shared on Snapchat are not managed? Is there a special way to handle the snapchat link I will miss? Or is it a bug with snapchat and there is a "workaround"?

Thank you in advance for your reading,
if you have any questions, I will try to answer them as best as I can.

**Comment by mikehardy** [comment, marked as answer]
With apologies - this will have to go upstream to either (or both of!) firebase-android-sdk and firebase-ios-sdk on github. There's nothing in react-native-firebase that will affect this, but the underlying SDKs definitely could, and if you tag me in there I can also try to help. The bug report looks amazing, just pasting it in there as a starting point should work

**Comment by sahailee** [comment]
Hi just stumbled on this issue as well, I'm working with firebase on iOS and android. Any luck finding a solution @vikjovanov?

**Reply by vikjovanov** [reply]
Hey!

In fact, this is due to the applications' webview!
Some of them (like Facebook, Messenger, ...) manage dynamic links (AppLink on Android) while Snapchat's webview works in full deep linking! So I followed the tutorial [here](https://firebase.google.com/docs/dynamic-links/android/receive) in order to install both the dynamics links management AND the plain deep linking management. 
The two points that really helped me to understand were "Add an intent filter for deep links" and "Handling Dynamic Links using App Links", by respecting these two points, I was able to set up the dynamic links management and the plain deep linking management.

And since then, it works great on Snapchat.

I hope I've answered your question!

**Reply by sahailee** [reply]
Thank you so much! 😀

**Reply by benedelstein** [reply]
@vikjovanov the tutorial you linked only applies for android, did you find a solution for iOS as well?
